opal_client/data/updater.py

```diff
@@ -233,5 +203,83 @@ async def wait_until_done(self):
         if self._subscriber_task is not None:
             await self._subscriber_task
 
+    @staticmethod
+    def calc_hash(data):
+        if not isinstance(data, str):
+            data = json.dumps(data)
+        return hashlib.sha256(data.encode('utf-8')).hexdigest()
```

Comment:
maybe in some scenarios can fail and throw?
what if data is None?

Reply:
Will fix

Reply:
Actually None will just be converted to null by JSON dumps.
I'll add a safeguard try/catch that will return an empty string and log just in case